tests/dialects/test_scf.py

```diff
@@ -56,6 +57,352 @@ def test_parallel():
+    rro = ReduceReturnOp.get(reduce_constant)
+    reduce_block = Block(arg_types=[i32, i32])
+    reduce_block.add_ops([reduce_constant, rro])
+
+    b.add_op(ReduceOp.get(init_val, reduce_block))
+    b.add_op(Yield.get())
+
+    body = Region(b)
+    p = ParallelOp.get([lbi], [ubi], [si], body, initVals)
+    # This should verify
+    p.verify()
+
+
+def test_parallel_verify_reduction_and_block_type_fails():
+    lbi = Constant.from_int_and_width(0, IndexType())
+    ubi = Constant.from_int_and_width(10, IndexType())
+    si = Constant.from_int_and_width(1, IndexType())
+
+    init_val = Constant.from_int_and_width(10, i64)
+
+    initVals = [init_val]
+
+    b = Block(arg_types=[IndexType()])
+
+    reduce_constant = Constant.from_int_and_width(100, i32)
+    rro = ReduceReturnOp.get(reduce_constant)
+    reduce_block = Block(arg_types=[i32, i32])
+    reduce_block.add_ops([reduce_constant, rro])
+
+    b.add_op(ReduceOp.get(init_val, reduce_block))
+    b.add_op(Yield.get())
+
+    body = Region(b)
+    p = ParallelOp.get([lbi], [ubi], [si], body, initVals)
+    with pytest.raises(VerifyException):
+        p.verify()
+
+
+def test_parallel_verify_yield_last_op():
+    lbi = Constant.from_int_and_width(0, IndexType())
+    ubi = Constant.from_int_and_width(10, IndexType())
+    si = Constant.from_int_and_width(1, IndexType())
+
+    b = Block(arg_types=[IndexType()])
+    b.add_op(Yield.get())
+
+    body = Region(b)
+    p = ParallelOp.get([lbi], [ubi], [si], body)
+    # This should verify
+    p.verify()
+
+    p2 = ParallelOp.get([lbi], [ubi], [si], Region(Block(arg_types=[IndexType()])))
+    with pytest.raises(VerifyException):
+        p2.verify()
+
+    b3 = Block(arg_types=[IndexType()])
+    b3.add_op(Constant.from_int_and_width(1, IndexType()))
+    p3 = ParallelOp.get([lbi], [ubi], [si], Region(b3))
+    with pytest.raises(VerifyException):
+        p3.verify()
+
+
+def test_parallel_verify_yield_zero_ops():
+    lbi = Constant.from_int_and_width(0, IndexType())
+    ubi = Constant.from_int_and_width(10, IndexType())
+    si = Constant.from_int_and_width(1, IndexType())
+
+    val = Constant.from_int_and_width(10, i64)
+
+    b = Block(arg_types=[IndexType()])
+    b.add_op(Yield.get(val))
+    body = Region(b)
+    p = ParallelOp.get([lbi], [ubi], [si], body)
+    with pytest.raises(VerifyException):
+        p.verify()
+
+    b2 = Block(arg_types=[IndexType()])
+    b2.add_op(Yield.get())
+    body2 = Region(b2)
+    p2 = ParallelOp.get([lbi], [ubi], [si], body2)
+    # Should verify as yield is empty
+    p2.verify()
+
+
+def test_parallel_test_count_number_reduction_ops():
+    reductions: List[ReduceOp] = []
+    for i in range(10):
+        init_val = Constant.from_int_and_width(i, i32)
+        reduce_op = ReduceOp.get(init_val, Block())
+        reductions.append(reduce_op)
+
+    b = Block()
+    b.add_ops(reductions)
+    p = ParallelOp.get([], [], [], Region(b))
```

Comment:
```suggestion
    @Builder.implicit_region
    def body():
        for i in range(10):
            init_val = Constant.from_int_and_width(i, i32)
            ReduceOp.get(init_val, Block())
p = ParallelOp.get([], [], [], body)
```
This should create exactly the same IR

Reply:
Similarly below

Reply:
Thanks, that's great - fixed :)